File: LeetCode/1343/main.py
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        n = len(arr)
        s = sum(arr[0:k-1])
        left, ans = 0, 0

        for i in range(k-1, n, 1) :

            s += arr[i]
            avg = s / k

            if(avg >= threshold) :
                ans += 1
            
            s -= arr[left]
            left += 1

        return ans
            

File: LeetCode/1343/main.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        int n = arr.size();
        int left = 0, ans = 0, s = 0;

        for(int i = 0; i < k; i++) {
            s += arr[i];
        }

        for(int i = k + 1; i < n; i++) {
            s += arr[i];
            if(s / k >= threshold) {
                ans++;
            }
            s -= arr[left];
            left++;
        }
        return ans;
    }
};